anime_downloader/sites/helpers/selescrape.py

```diff
@@ -85,113 +128,115 @@ def driver_select(): #
-    try:
-        if status == 'hide':
-            driver.get(url)
-        elif status == 'show':
-            r = requests.head(url)
-            if r.status_code == 503:
-                raise RuntimeError("This website's sevice is unavailable or has cloudflare on.")
-            driver.get(url)
-            return r.status_code
-        else:
-            driver.get(url)
-    except requests.ConnectionError:
-        raise RuntimeError("Failed to establish a connection using the requests library.")
-
-
 def cloudflare_wait(driver):
     '''
     It waits until cloudflare has gone away before doing any further actions.
-    The way it works is by getting the title of the page 
+    The way it works is by getting the title of the page
     and as long as it is "Just a moment..." it will keep waiting.
-    This part of the code won't make the code execute slower 
-    if the target website has not a Cloudflare redirection.
-    At most it will sleep 1 second as a precaution. 
-    Also, i have made it time out after 30 seconds, useful if the target website is not responsive 
+    This part of the code won't make the code execute slower
+    if the target website has no Cloudflare redirection.
+    At most it will sleep 1 second as a precaution.
+    Also, i have made it time out after 50 seconds, useful if the target website is not responsive
     and to stop it from running infinitely.
     '''
-    abort_after = 30
+    abort_after = 50
     start = time.time()
 
     title = driver.title  # title = "Just a moment..."
     while title == "Just a moment...":
         time.sleep(0.25)
         delta = time.time() - start
         if delta >= abort_after:
-            logger.error(f'Timeout:\nCouldnt bypass cloudflare. \
-            See the screenshot for more info:\n{get_data_dir()}/screenshot.png')
+            logger.error(f'Timeout:\tCouldnt bypass cloudflare. \
+            See the screenshot for more info:\t{get_data_dir()}/screenshot.png')
+            break
         title = driver.title
         if not title == "Just a moment...":
             break
-    time.sleep(1) # This is necessary to make sure everything has loaded fine.
+    time.sleep(1)  # This is necessary to make sure everything has loaded fine.
 
 
-def request(request_type, url, **kwargs): #Headers not yet supported , headers={}
+def request(request_type, url, **kwargs):  # Headers not yet supported , headers={}
     params = kwargs.get('params', {})
-    new_url = add_url_params(url, params)
-    driver = driver_select()
-    status = status_select(driver, new_url, 'hide')
-    try:
-        cloudflare_wait(driver)
-        user_agent = driver.execute_script("return navigator.userAgent;") #dirty, but allows for all sorts of things above
-        cookies = driver.get_cookies()
-        text = driver.page_source
-        driver.close()
+    url = url if not params else url + '?' + urlencode(params)
+
+    if bool(check_cache(url)):
+        cached_data = check_cache(url)
```

Comment:
Instead of running the method twice on the same url, perhaps store in a variable and check the variable?